src/OpenDebugAD7/Tracepoint.cs

```diff
@@ -68,13 +68,14 @@ private int GetInterpolatedLogMessage(string logMessage, IDebugThread2 pThread,
             hr = frameInfoEnum.Next(1, topFrame, ref fetched);
             if (hr < 0 || fetched != 1 || topFrame[0].m_pFrame == null)
             {
-                message = string.Format(CultureInfo.InvariantCulture, "Unable to interpolate logMessage because there is no top frame.");
+                message = AD7Resources.Error_InterpolateMissingTopFrame;
                 return hr;
             }
 
             Dictionary<string, string> seenExpressions = new Dictionary<string, string>();
             StringBuilder sb = new StringBuilder();
             int currIndex = 0;
+            bool failureSeen = false;
```

Comment:
why not just create int hr = Constants.S_OK and set it to Constants.E_FAIL in your bool area and return hr at the bottom?